File: symfony-app/src/Entity/Accompagnement.php
<?php

namespace App\Entity;

use App\Repository\AccompagnementRepository;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity(repositoryClass: AccompagnementRepository::class)]
#[ORM\Table(name: 'accompagnements')]
class Accompagnement
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column(type: 'integer')]
    private ?int $id = null;

    #[ORM\ManyToOne(targetEntity: Abonnement::class, inversedBy: 'accompagnements')]
    #[ORM\JoinColumn(name: 'abonnement_id', nullable: false, onDelete: 'CASCADE')]
    private Abonnement $abonnement;

    #[ORM\Column(type: 'string', length: 100)]
    private string $nom;

    #[ORM\Column(type: 'text', nullable: true)]
    private ?string $description = null;

    #[ORM\Column(type: 'date')]
    private \DateTimeInterface $dateDebut;

    #[ORM\Column(type: 'date', nullable: true)]
    private ?\DateTimeInterface $dateFin = null;

    #[ORM\Column(type: 'string', length: 50, nullable: true)]
    private ?string $typeAccompagnement = null;

    #[ORM\Column(type: 'string', length: 20, options: ['default' => 'en_cours'])]
    private string $statut = 'en_cours';

    #[ORM\Column(type: 'datetime', options: ['default' => 'CURRENT_TIMESTAMP'])]
    private \DateTimeInterface $createdAt;

    #[ORM\Column(type: 'datetime', options: ['default' => 'CURRENT_TIMESTAMP'])]
    private \DateTimeInterface $updatedAt;

    #[ORM\OneToMany(mappedBy: 'accompagnement', targetEntity: PlanExercice::class)]
    private Collection $plansExercices;

    #[ORM\OneToMany(mappedBy: 'accompagnement', targetEntity: PlanRegime::class)]
    private Collection $plansRegimes;

    public function __construct()
    {
        $this->createdAt = new \DateTimeImmutable();
        $this->updatedAt = new \DateTimeImmutable();
        $this->plansExercices = new ArrayCollection();
        $this->plansRegimes = new ArrayCollection();
    }

    public function getId(): ?int { return $this->id; }
    public function getAbonnement(): Abonnement { return $this->abonnement; }
    public function setAbonnement(Abonnement $abonnement): void { $this->abonnement = $abonnement; }
    public function getNom(): string { return $this->nom; }
    public function setNom(string $nom): void { $this->nom = $nom; }
    public function getDescription(): ?string { return $this->description; }
    public function setDescription(?string $description): void { $this->description = $description; }
    public function getDateDebut(): \DateTimeInterface { return $this->dateDebut; }
    public function setDateDebut(\DateTimeInterface $dateDebut): void { $this->dateDebut = $dateDebut; }
    public function getDateFin(): ?\DateTimeInterface { return $this->dateFin; }
    public function setDateFin(?\DateTimeInterface $dateFin): void { $this->dateFin = $dateFin; }
    public function getTypeAccompagnement(): ?string { return $this->typeAccompagnement; }
    public function setTypeAccompagnement(?string $typeAccompagnement): void { $this->typeAccompagnement = $typeAccompagnement; }
    public function getStatut(): string { return $this->statut; }
    public function setStatut(string $statut): void { $this->statut = $statut; }
    public function getCreatedAt(): \DateTimeInterface { return $this->createdAt; }
    public function setCreatedAt(\DateTimeInterface $createdAt): void { $this->createdAt = $createdAt; }
    public function getUpdatedAt(): \DateTimeInterface { return $this->updatedAt; }
    public function setUpdatedAt(\DateTimeInterface $updatedAt): void { $this->updatedAt = $updatedAt; }
    public function getPlansExercices(): Collection { return $this->plansExercices; }
    public function getPlansRegimes(): Collection { return $this->plansRegimes; }
}

File: javafx-app/src/main/java/com/santea/model/Accompagnement.java
package com.santea.model;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

public class Accompagnement {
    private Integer id;
    private Abonnement abonnement;
    private String nom;
    private String description;
    private LocalDateTime dateDebut;
    private LocalDateTime dateFin;
    private String typeAccompagnement;
    private String statut;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    private List<Object> plansExercices;
    private List<Object> plansRegimes;

    public Accompagnement() {
        this.plansExercices = new ArrayList<>();
        this.plansRegimes = new ArrayList<>();
    }

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public Abonnement getAbonnement() {
        return abonnement;
    }

    public void setAbonnement(Abonnement abonnement) {
        this.abonnement = abonnement;
    }

    public String getNom() {
        return nom;
    }

    public void setNom(String nom) {
        this.nom = nom;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public LocalDateTime getDateDebut() {
        return dateDebut;
    }

    public void setDateDebut(LocalDateTime dateDebut) {
        this.dateDebut = dateDebut;
    }

    public LocalDateTime getDateFin() {
        return dateFin;
    }

    public void setDateFin(LocalDateTime dateFin) {
        this.dateFin = dateFin;
    }

    public String getTypeAccompagnement() {
        return typeAccompagnement;
    }

    public void setTypeAccompagnement(String typeAccompagnement) {
        this.typeAccompagnement = typeAccompagnement;
    }

    public String getStatut() {
        return statut;
    }

    public void setStatut(String statut) {
        this.statut = statut;
    }

    public LocalDateTime getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(LocalDateTime createdAt) {
        this.createdAt = createdAt;
    }

    public LocalDateTime getUpdatedAt() {
        return updatedAt;
    }

    public void setUpdatedAt(LocalDateTime updatedAt) {
        this.updatedAt = updatedAt;
    }

    public List<Object> getPlansExercices() {
        return plansExercices;
    }

    public void setPlansExercices(List<Object> plansExercices) {
        this.plansExercices = plansExercices;
    }

    public List<Object> getPlansRegimes() {
        return plansRegimes;
    }

    public void setPlansRegimes(List<Object> plansRegimes) {
        this.plansRegimes = plansRegimes;
    }
}
